scripts/utils/install_detect_secrets.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+[ ! -f "$HOME/.default-python-packages" ] && touch "$HOME/.default-python-packages"
+
+# Array of pip packages to be added
+pip_packages=("virtualenv" "pre-commit" "detect-secrets==1.5.0")
+for package in "${pip_packages[@]}"; do
+  if ! grep -q "$package" "$HOME/.default-python-packages"; then
+    echo "$package" >> "$HOME/.default-python-packages"
+  fi
+done
+
+echo "legacy_version_file = yes" >> "$HOME/.asdfrc"
+asdf plugin add python
```

Comment:
Related to https://github.com/artsy/eigen/pull/11878#discussion_r2037391113

I just remembered that this command is not backwards compatible https://asdf-vm.com/guide/upgrading-to-v0-16.html#hyphenated-commands-have-been-removed

That means if someone with `asdf` version < `0.16.0` is running this script, it will fail to add the plugin. If this script is aiming to be backwards compatible with `asdf` versions prior to `0.16.0` then:
```suggestion
if asdf plugin add --help &>/dev/null; then
  asdf plugin add python
else
  asdf plugin-add python
fi
```

Reply:
```
~/C/eigen (brian+george/expo-again|✔) [1] $ asdf plugin add python                                                                          (base) 
updating plugin repository...HEAD is now at c0369a1 feat: add yasm plugin (#1087)
Plugin named python already added
~/C/eigen (brian+george/expo-again|✔) $ asdf --version                                                                                      (base) 
v0.15.0
``` 
hmm this actually works on my version, maybe earlier versions it didn't, but i don't see any harm in checking